Controllers/EmployeeManagement/EmployeesController.cs

```diff
@@ -95,29 +104,61 @@ public ActionResult<Employee> Post([FromBody] Employee value)
         /// <param name="value">changed employee</param>
         /// <returns>one employee by id</returns>
         /// <response code="204">if the change is successful</response>
-        /// <response code="404">if bad request</response>
+        /// <response code="400">if bad request</response>
         // PUT api/<EmployeesController>/5
         [HttpPut]
-        public ActionResult<Employee> Put(int id, [FromBody] Employee value)
+        public async Task<ActionResult<Employee>> Put(int id, [FromBody] Employee value)
         {
-            if (id != value.id)
+            var employee = await _context.Employees.FindAsync(id);
+            var businessAdmin = await _context.BusinessAdministrators.FindAsync(value.CreatedBy);
+
+            if (employee == null)
+            {
+                return BadRequest("No such employee with the given id");
+            }
+
+            if (businessAdmin == null)
             {
-                return BadRequest();
+                return BadRequest("No such business admin with the given CreatedBy value");
             }
 
+            value.id = id;
+
+            var response = Employee.CheckIfValid(value);
+
+            if (!response.Equals("Ok"))
+            {
+                return BadRequest("Invalid body: " + response);
+            }
+
+            _context.ChangeTracker.Clear();
+            _context.Employees.Update(value);
+            _context.SaveChanges();
+
             return NoContent();
         }
 
         /// <summary>
         /// Deletes specific employee by id from the employees table
         /// </summary>
+        /// 
         /// <param name="id">employee id</param>
         /// <response code="204">if delete is successful</response>
         // DELETE api/<EmployeesController>/5
         [HttpDelete("{id}")]
         public ActionResult Delete(int id)
```

Comment:
Suggestion: add isDeleted field to Employee and don't do full delete if there are some Orders associated with them, so that the orders don't get cascade deleted as well. I did something like that with [ProductServices](https://github.com/EmilijaSvirsk/PoSS_Implementation/pull/4/files#diff-38663edad61bbc23da9f598846021c3d162e6a3271bbc8c068a2d7d0cbf9bf66R121). 

This is not mandatory but it would be nice :).  If you decide to do it don't forget to hide isDeleted field in Get, Post, and Put; and also filter by !employee.isDeleted